tinygrad/tensor.py

```diff
@@ -3129,6 +3129,20 @@ def div(self, x:Union[Tensor, ConstType], reverse=False) -> Tensor:
     numerator, denominator = self._broadcasted(x, reverse)
     return numerator.cast(least_upper_float(numerator.dtype)) * denominator.cast(least_upper_float(denominator.dtype)).reciprocal()
 
+  def mod(self, x:Union[Tensor, ConstType], reverse=False) -> Tensor:
+    """
+    Mod `self` by `x`.
+    Equivalent to `self % x`.
+    Supports broadcasting to a common shape, type promotion, and integer inputs.
+
+    ```python exec="true" source="above" session="tensor" result="python"
+    print(Tensor([-4, 7, 5, 4, -7, 8]).mod(Tensor([2, -3, 8, -2, 3, 5])).numpy())
+    ```
+    """
+    a, b = self._broadcasted(x, reverse)
+    r = F.Mod.apply(a, b).contiguous()
```

Comment:
Why is this contiguous needed?

Reply:
https://github.com/chenyuxyz/tinygrad/pull/636
without it tests fail.

`(UPat(GroupOp.ALU, name="x"), lambda x: x.const_like(x.vmin) if x.vmin == x.vmax else None)`
This is the rule that's causing problems.